### PR TITLE
[Bugfix][Triton] Fix large-stride KV address overflow in paged MQA logits

### PR DESCRIPTION
**Pull request by shen-shanshan**
## Motivation

Fix `deepgemm_fp8_paged_mqa_logits` producing non-finite logits when paged KV cache blocks are accessed through a large physical block stride.

- Bug report: https://github.com/Fangzhou-Ai/vllm/issues/13
- vLLM side temporary fix: https://github.com/vllm-project/vllm/pull/49714

This is distinct from #4244. That change fixed 32-bit `buffer_store` output row offsets. This change fixes 32-bit KV value/scale input load offsets.

### Root Cause

The affected vLLM sparse-attention cache is a per-layer view over a larger allocation. A representative failing call had:

- KV block stride: `1,435,968` bytes
- Maximum referenced physical block: `4564`
- Maximum KV byte offset: `6,553,757,952`

The wrapper declared `stride_k_seq` and `stride_scale_seq` as `i32`. The preshuffle Gluon kernel then formed AMD `buffer_load` offsets as:

```text
physical_block_id * stride_k_seq + in_block_offset
```

AMD buffer voffsets are 32-bit. The multiplication overflowed for the real cache layout, causing the kernel to read unrelated FP8 values/scales. Those loads produced `NaN` and `+Inf` logits even though all referenced inputs were finite and all output positions were written.

Runtime instrumentation observed 9 NaNs and 1 positive infinity among 3,748 valid logits in a failing call. Q, weights, referenced KV values/scales, context lengths, and block-table entries were all valid. The same corruption appeared at identical coordinates on all eight TP ranks.

Compacting the exact same referenced KV blocks removed all non-finite outputs, isolating the failure to large-stride addressing.

### Fix Method

- Declare `stride_k_seq` and `stride_scale_seq` as `i64` in the compiled kernel signature.
- Use 64-bit pointer arithmetic for preshuffled KV value and scale loads in the exercised pipeline branch.
- Use `gl.load` for these accesses because Gluon AMD `buffer_load` requires an `int32`/`uint32` offset tensor and therefore cannot represent the failing address range.
- Add a regression test whose KV layout crosses the 32-bit byte-offset boundary and compare it against an identical compact-layout reference.

### Exact captured-input replay

The failing input was replayed with the original block stride, physical block IDs, and a maximum byte offset of `6,553,757,952`.

### End-to-end

As mentioned in https://github.com/Fangzhou-Ai/vllm/issues/13.

### Regression test

```bash
python -m pytest -sv op_tests/test_pa_mqa_logits_offset.py::test_paged_mqa_logits_large_kv_stride_matches_compact_reference
```

The test constructs identical compact and large-stride paged KV inputs, forces the physical KV address beyond `2**32`, and requires finite, bit-identical logits.

### Exact captured-input replay

- Before: 10 non-finite valid logits
- After: 0 non-finite logits and 0 unwritten sentinel values

### End-to-end

The original concurrent DeepSeek-V4 TP8 serving workload completed without non-finite paged-MQA logits or a GPU memory access fault after the fix.

### Regression test

```text
1 passed in 16.92s
```

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

**Comment by shen-shanshan** [comment]
Fixed in https://github.com/ROCm/aiter/pull/4774.
